internal/controller/object/object.go

```diff
@@ -348,6 +356,8 @@ func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 	}, current)
 
 	if kerrors.IsNotFound(err) {
+		// remove cache entry
+		c.desiredStateCacheManager.Remove(mg)
```

Comment:
Not sure whether mg would have UID field set in case of a not found error returned from the get request 🤔 
We may choose to move this to Delete().

Reply:
done, see https://github.com/turkenh/provider-kubernetes/pull/1/commits/9b24ebfeb53f388bb31c164874c31aa71d0167dd